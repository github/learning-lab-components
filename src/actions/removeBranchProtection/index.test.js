const removeBranchProtection = require('.')
const mockContext = require('../../../tests/mockContext')

describe('removeBranchProtection', () => {
  let context

  beforeEach(() => {
    context = mockContext({}, {
      repos: { removeBranchProtection: jest.fn() }
    })
  })

  it('removes branch protection on master by default', async () => {
    await removeBranchProtection(context, {})
    expect(context.github.repos.removeBranchProtection).toHaveBeenCalled()
    expect(context.github.repos.removeBranchProtection.mock.calls).toMatchSnapshot()
  })

  it('removes branch protection on the provided branch', async () => {
    await removeBranchProtection(context, { branch: 'a-branch' })
    expect(context.github.repos.removeBranchProtection).toHaveBeenCalled()
    expect(context.github.repos.removeBranchProtection.mock.calls).toMatchSnapshot()
  })
})
