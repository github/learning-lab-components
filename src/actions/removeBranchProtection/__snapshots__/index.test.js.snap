// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeBranchProtection removes branch protection on master by default 1`] = `
Array [
  Array [
    Object {
      "branch": "master",
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;

exports[`removeBranchProtection removes branch protection on the provided branch 1`] = `
Array [
  Array [
    Object {
      "branch": "a-branch",
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;
